keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,731 @@
+    // Driver specifies the name of the DRA driver whose kubelet
+    // plugin should be invoked to process the allocation once the claim is
+    // needed on a node.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
+    //
+    // +required
+    Driver string `json:"driver" protobuf:"bytes,1,rep,name=driver"`
+
+    // This name together with the driver name and the device name field
+    // identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+    //
+    // Must not be longer than 253 characters and may contain one or more
+    // DNS sub-domains separated by slashes.
+    //
+    // +required
+    Pool string `json:"pool" protobuf:"bytes,2,rep,name=pool"`
+
+    // Device references one device instance via its name in the driver's
+    // resource pool. It must be a DNS label.
+    //
+    // +required
+    Device string `json:"device" protobuf:"bytes,3,rep,name=device"`
+
+    // Conditions contains the latest observation of the device's state.
+    // If the device has been configured according to the class and claim
+    // config references, the `Ready` condition should be True.
+    //
+    // +optional
+    // +listType=atomic
+    Conditions []metav1.Condition `json:"conditions" protobuf:"bytes,4,opt,name=conditions"`
+
+    // Data contains arbitrary driver-specific data.
+    //
+    // +optional
+    Data runtime.RawExtension `json:"data,omitempty" protobuf:"bytes,5,opt,name=data"`
+
+    // NetworkData contains network-related information specific to the device.
+    //
+    // +optional
+    NetworkData NetworkDeviceData `json:"networkData,omitempty" protobuf:"bytes,6,opt,name=networkData"`
+}
+
+// NetworkDeviceData provides network-related details for the allocated device.
+// This information may be filled by drivers or other components to configure
+// or identify the device within a network context.
+type NetworkDeviceData struct {
+    // InterfaceName specifies the name of the network interface associated with
+    // the allocated device. This might be the name of a physical or virtual
+    // network interface.
+    //
+    // +optional
+    InterfaceName string `json:"interfaceName,omitempty" protobuf:"bytes,1,opt,name=interfaceName"`
+
+    // Addresses lists the network addresses assigned to the device's network interface.
+    // This can include both IPv4 and IPv6 addresses.
+    // The addresses are in the CIDR notation, which includes both the address and the
+    // associated subnet mask.
+    // e.g.: "192.0.2.0/24" for IPv4 and "2001:db8::/64" for IPv6.
+    //
+    // +optional
+    // +listType=atomic
+    Addresses []string `json:"addresses,omitempty" protobuf:"bytes,2,opt,name=addresses"`
+
+    // HWAddress represents the hardware address (e.g. MAC Address) of the device's network interface.
+    //
+    // +optional
+    HWAddress string `json:"hwAddress,omitempty" protobuf:"bytes,3,opt,name=hwAddress"`
```

Comment:
So the use case for including IP addresses in the result is that people may want to connect to the pod via this interface. That's something someone might want to do even if they aren't directly poking around with the allocated device.

What is the use case for including HW addresses? What sort of user would have a reason to need to know the hwaddr, but would _not_ be doing something where they had to look directly at the allocated device in question (which would allow them to find the hwaddr themselves)?

Reply:
I am not directly involved any more, but I think non-standard (even seemingly weird) use-cases that don't use ip-addresses should be supported. An example is non-IP traffic, another is to allow "something" to send ip-packets to the interface even when it has no ip-address, or has an address shared by many.

The entity that consumes this status will probably not be an application that wants to connect to the pod, but something very specialized (router, load-balancer, ...?), and we can't predict what they will be.